pandas/core/groupby/ops.py

```diff
@@ -451,11 +451,7 @@ def _cython_operation(
 
         # categoricals are only 1d, so we
         # are not setup for dim transforming
-        if is_categorical_dtype(values) or is_sparse(values):
```

Comment:
This was the "fix" for #29746 - not sure this is really the right way to do that but makes tests green for now

Reply:
yah this definitely looks sketchy